### PR TITLE
Store single settlement instead of HashMap

### PR DESCRIPTION
**Pull request by vkgnosis**
Follow up to https://github.com/cowprotocol/services/pull/1056#discussion_r1068705084 . The api documents that /solve and /execute must be called after each other for the same settlement so there is no need to use a HashMap. This makes this code simpler because there is no background task and no concurrent HashMap.

### CODE REVIEW

**Comment by vkgnosis** [comment]
You preferred merging this after your first PR is merged, right? So I won't merge it yet. Or if you do want to merge it immediately, feel free to press the button.

**Review by MartinquaXD — APPROVED** [review]

File: crates/driver/src/domain/competition/mod.rs
Comment (vkgnosis) [reply crates/driver/src/domain/competition/mod.rs:69]: I assumed requests to these endpoints are always authenticated so there should be no malicious calls. An erroneous call is a bug in autopilot. I don't think it makes sense to workaround a potential bug this way. If it happened we would fix it immediately.

File: crates/driver/src/domain/competition/mod.rs
Comment (vkgnosis) [reply crates/driver/src/domain/competition/mod.rs:77]: Not part of this PR but yeah it's a potential follow up task. The proper way to handle restarts would probably to wait until the current auction is over. Then there is no need for persistence unless you are worried about crashes.

File: crates/driver/src/domain/competition/mod.rs
Comment (nlordell) [reply crates/driver/src/domain/competition/mod.rs:69]: I think `take` here makes slightly more sense in my mental model - you probably don't want duplicate executions for the same settlement to be floating around in the `mempool`.

**Review by nlordell — APPROVED** [review]

File: crates/driver/src/domain/competition/mod.rs
Comment (MartinquaXD) [reply crates/driver/src/domain/competition/mod.rs:77]: I was more concerned with crashes.

Comment (nlordell) [reply crates/driver/src/domain/competition/mod.rs:77]: I agree that we should be resistant to crashes. I also agree that this makes more sense in a separate PR. @vkgnosis - do you mind creating a GH issue so we don't forget to get back to this?

Comment (vkgnosis) [reply crates/driver/src/domain/competition/mod.rs:77]: https://github.com/cowprotocol/services/issues/1092

**Comment by vkgnosis** [comment]
Feel free to do anything necessary to merge it.

**Pull request merged**
